signer-npm/js/src/utils.js

```diff
@@ -51,10 +52,17 @@ function addressAsBytes(address) {
     payload,
     checksum;
   const protocolIndicator = address[1];
+  const protocolIndicatorByte = `0${protocolIndicator}`;
 
   switch (Number(protocolIndicator)) {
     case ProtocolIndicator.ID:
-      throw new ProtocolNotSupported("ID");
+      if (address.length > 18) {
+        throw new InvalidPayloadLength();
+      }
+      return Buffer.concat([
+        Buffer.from(protocolIndicatorByte, "hex"),
+        Buffer.from(leb.unsigned.encode(address.substr(2)))
```

Comment:
Can address be shorter and result in errors?

Reply:
It can't be shorter because right before we verify that the length is minimum 18 : 
```
if (address.length > 18) {
```
Which if it is smaller will throw an error

Reply:
My bad you are right I misread the length check